src/pretix/api/serializers/event.py

```diff
@@ -15,14 +17,73 @@ def to_representation(self, value):
         }
 
 
+class PluginsField(Field):
+
+    def to_representation(self, obj):
+        from pretix.base.plugins import get_all_plugins
+
+        plugins = {
+            p.module for p in get_all_plugins()
+            if not p.name.startswith('.') and getattr(p, 'visible', True) and p.module in obj.get_plugins()
+        }
+
+        return plugins
+
+    def to_internal_value(self, data):
+        from pretix.base.plugins import get_all_plugins
+
+        plugins_available = {
+            p.module for p in get_all_plugins()
+            if not p.name.startswith('.') and getattr(p, 'visible', True)
+        }
+
+        for plugin in data:
+            if plugin not in plugins_available:
+                raise ValidationError(
+                    message=_("Unknown plugin: '%s'."),
+                    params=(plugin,)
+                )
+                break
+
+        plugins = {plugin_name for plugin_name in data}
+
+        return {
+            'plugins': ",".join(plugins)
+        }
+
+
 class EventSerializer(I18nAwareModelSerializer):
-    meta_data = MetaDataField(source='*')
+    plugins = PluginsField(required=False, source='*')
 
     class Meta:
         model = Event
         fields = ('name', 'slug', 'live', 'currency', 'date_from',
                   'date_to', 'date_admission', 'is_public', 'presale_start',
-                  'presale_end', 'location', 'has_subevents', 'meta_data')
+                  'presale_end', 'location', 'has_subevents', 'meta_data', 'plugins')
+
+    def validate(self, data):
+        data = super().validate(data)
+
+        full_data = self.to_internal_value(self.to_representation(self.instance)) if self.instance else {}
+        full_data.update(data)
+
+        Event.clean_dates(data.get('date_from'), data.get('date_to'))
+        Event.clean_presale(data.get('presale_start'), data.get('presale_end'))
+
+        return data
+
+    def validate_slug(self, value):
+        Event.clean_slug(self.context['request'].organizer, self.instance, value)
```

Comment:
We also need to make sure that the slug cannot be changed once it has been set.

Reply:
Implemented in `Event.clean_slug` and covered by `test_events.py:L275`

Reply:
Thanks!